File: r/exploratory_analysis.R
# Independent portfolio case study using synthetic data inspired by common photography-business workflows.
d <- read.csv("data/processed/analytics_dataset.csv")
stopifnot(all(d$amount_paid >= 0), nrow(d) > 0)
monthly <- aggregate(amount_paid ~ month, d, sum)
png("visuals/r_monthly_revenue.png",900,500); plot(monthly$amount_paid,type="l",col="#2F6F73",lwd=3,xaxt="n",main="Monthly Revenue (R validation)",ylab="USD",xlab="Month"); axis(1,seq_len(nrow(monthly)),monthly$month,las=2,cex.axis=.65); dev.off()

File: python/exploratory_analysis.py
"""Create separate recruiter-ready analytical figures."""
from pathlib import Path
import matplotlib.pyplot as plt
import pandas as pd
ROOT=Path(__file__).resolve().parents[1]; df=pd.read_csv(ROOT/"data"/"processed"/"analytics_dataset.csv")
plt.rcParams.update({"figure.figsize":(10,5.5),"axes.titleweight":"bold","axes.spines.top":False,"axes.spines.right":False})
def save(series,title,ylabel,file,kind="bar"):
    fig,ax=plt.subplots(); getattr(series.plot,kind)(ax=ax,color="#C26A4A" if kind=="bar" else "#2F6F73"); ax.set_title(title); ax.set_ylabel(ylabel); ax.grid(axis="y",alpha=.2); fig.tight_layout(); fig.savefig(ROOT/"visuals"/file,dpi=160); plt.close(fig)
monthly=df.groupby("month").amount_paid.sum(); save(monthly,"Monthly Revenue — Synthetic Data","Revenue (USD)","monthly_revenue.png","line")
lead=df.groupby("lead_source").agg(leads=("lead_id","count"),bookings=("booking_id","count")).sort_values("bookings"); save(lead.bookings,"Bookings by Lead Source","Bookings","booking_conversion.png")
pkg=df.groupby("package_name").amount_paid.sum().sort_values(); save(pkg,"Revenue by Package","Revenue (USD)","package_performance.png")
status=df.booking_status.value_counts(); save(status,"Booking Outcomes","Bookings","cancellation_rate.png")
save(lead.bookings,"Lead Source Performance","Bookings","lead_source_performance.png")

